### PR TITLE
SuiteSparse: update to 5.13.0

### PR DESCRIPTION
**Pull request by szhorvat**
- update to 5.13.0
 - fix SuiteSparse_SLIP_LU version (should have been 1.0.2)
 - ref: https://trac.macports.org/ticket/67084

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 13.6 22G120 arm64
Xcode 15.0 15A240d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by szhorvat** [comment]
Updates SuiteSparse to 5.13.0 as proposed in https://trac.macports.org/ticket/67084 The latest version 7.2.1, but it's 5.13.0 should have no breaking changes, thus a gradual update path is better.

Most sub-ports do not change. The major update compared to thee previous version in MacPorts is GraphBLAS.

I am aware of the `port lint` errors, but I believe these are false positives, and they are not due to the changed in this PR.

**Comment by szhorvat** [comment]
Strange that the macOS 13 CI fails. It worked on my local 13.6 arm64 system. Error in CI is:

```
2023-10-16T20:47:33.6254510Z ld: warning: ignoring duplicate libraries: '-lopenblas'
2023-10-16T20:47:33.6421700Z ld: duplicate LC_RPATH '/opt/local/lib/libgcc' in '/opt/local/lib/libopenblas-r1.dylib'
2023-10-16T20:47:33.6443060Z clang: error: linker command failed with exit code 1 (use -v to see invocation)
...
2023-10-16T20:47:33.6484310Z Error: Failed to build SuiteSparse_CHOLMOD: command execution failed
```

The actual linker error is not shown.

`SuiteSparse_CHOLMOD` is not actually change with this update.

**Comment by szhorvat** [comment]
See https://trac.macports.org/ticket/68487

**Comment by szhorvat** [comment]
Mysteriously, I could now reproduce the issue on my local machine, and it is fixable by reverting to Apple's older linker. I kept the fix in a separate commit to make it easier to revert once the linker is fixed (I believe this is a linker bug).

If the CI tests pass, I think this can be merged, @MarcusCalhoun-Lopez 

Also CC @barracuda156

**Comment by szhorvat** [comment]
Many ports that link to OpenBLAS are affected by this issue. It seems to me that the workaround should rather go into the `linalg` port group. For now I added the workaround to those SuiteSparse subports where it was necessary.

**Comment by szhorvat** [comment]
If PR #21127 is merged than the second commit within this PR becomes obsolete, and can be removed.

**Comment by szhorvat** [comment]
> If PR #21127 is merged than the second commit within this PR becomes obsolete, and can be removed.

I have now removed the second commit, since #21127 replaces it (CC @pmetzger)

**Comment by szhorvat** [comment]
Can we have this merged please? It's been over 4 days since the last push. I expect that the risk of breaking other packages is low, as most sub-ports did not change. The major update was the GraphBLAS subport.

**Comment by pmetzger** [comment]
Looks nearly good to go. Do you think you could have a look at all the 

```
Error: Line 153 repeats inclusion of PortGroup linear_algebra
Error: Line 176 repeats inclusion of PortGroup linear_algebra
```

errors from port lint?

**Comment by szhorvat** [comment]
Yes, these linter warnings were already present before this PR, and I believe they are false positives. This port is a bit unusual as it has so many subports which are essentially independent packages that just happen to be distributed together. Since these packages build independently, they have independent configurations, including independent port groups. We can't have a single inclusion of `linear_algebra`, as this port group sets up things that would not be appropriate for some of the subports.

In short, I think the port is fine, and the linter is wrong. I'm not sure how to get rid of the linter messages.

**Comment by szhorvat** [comment]
The force-push I just made did not change code. I just fixed the commit message:

"fix SuiteSparse_SLIP_LU (should have been 1.0.2)" --> "fix SuiteSparse_SLIP_LU version (should have been 1.0.2)"

File: math/SuiteSparse/Portfile [hunk 1]
```diff
@@ -188,7 +188,7 @@ subport SuiteSparse_SPQR {
 }
 
 subport SuiteSparse_SLIP_LU {
-    version                 2.0.9
+    version                 1.0.2
```
Comment (reneeotten) [inline math/SuiteSparse/Portfile:191]: shy is this version set back to an earlier version? If this is the reason for the "epoch" change then only set it here to "1" in this subport.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: math/SuiteSparse/Portfile (hunk 1 above)
Comment (reneeotten) [reply math/SuiteSparse/Portfile:191]: I see from the comment that there is no actual change here besides the version number. I would suggest not to change anything here and just update it when upstream does.

Comment (szhorvat) [reply math/SuiteSparse/Portfile:191]: @reneeotten Just to make sure we're on the same page: The version number in MacPorts was incorrect (copy-paste error from `SuiteSparse_SPQR`). The real version is 1.0.2, and it did not change between SuiteSparse 5.13 and the previous SuiteSparse available in MacPorts.

Upstream will never update this version. Instead, SuiteSparse 6.0 and later replaces SLIP_LU with SPEX.

Before I go ahead, can you please confirm that you still want me to leave the (incorrect) version alone?

Comment (szhorvat) [reply math/SuiteSparse/Portfile]: (In the meantime I pushed the change that I understand you are asking for, so we can make progress on this.)
